File: 프로그래머스/1/42576. 완주하지 못한 선수/완주하지 못한 선수.cpp
#include <iostream>
#include <string>
#include <vector>
#include <unordered_map>

using namespace std;

unordered_map<string, int> um;

string solution(vector<string> participant, vector<string> completion) {
    string answer = "";
    
    for (auto person : participant) {
        if(um[person]>0) {
            um[person]++;
        }
        else
            um[person] = 1;
    }
    
    for (auto person : completion) {
        um[person]--;
    }
    
    for (auto person : participant) {
        if(um[person]) {
            answer = person;
            return answer;
        }
    }
    
    
}

File: 프로그래머스/1/42576. 완주하지 못한 선수/완주하지 못한 선수.java
import java.util.*;
import java.util.stream.*;

class Solution {
    public String solution(String[] participant, String[] completion) {
        String answer = "";
        
        HashMap<String, Integer> map = new HashMap<>();
        
        for (String p : participant) {
            map.put(p, map.getOrDefault(p, 0) + 1);
        }
        
        for (String p : completion) {
            map.put(p, map.get(p) - 1);
        }
        
        List<String> list = map.keySet().stream().filter(key -> map.get(key) == 1).collect(Collectors.toList());
        
        return list.get(0);
    }
}